examples/model_selection/plot_cost_sensitive_learning.py

```diff
@@ -0,0 +1,774 @@
+# therefore need to define a small custom function to compute it.
+from sklearn.metrics import confusion_matrix
+
+
+def fpr_score(y, y_pred, neg_label, pos_label):
+    cm = confusion_matrix(y, y_pred, labels=[neg_label, pos_label])
+    tn, fp, _, _ = cm.ravel()
+    tnr = tn / (tn + fp)
+    return 1 - tnr
+
+
+# %%
+# As previously stated, the "positive label" is not defined as the value "1" and calling
+# some of the metrics with this non-standard value raise an error. We need to
+# provide the indication of the "positive label" to the metrics.
+#
+# We therefore need to define a scikit-learn scorer using
+# :func:`~sklearn.metrics.make_scorer` where the information is passed. We store all
+# the custom scorers in a dictionary. To use them, we need to pass the fitted model,
+# the data and the target on which we want to evaluate the predictive model.
+from sklearn.metrics import make_scorer, precision_score, recall_score
+
+tpr_score = recall_score  # TPR and recall are the same metric
+scoring = {
+    "precision": make_scorer(precision_score, pos_label=pos_label),
+    "recall": make_scorer(recall_score, pos_label=pos_label),
+    "fpr": make_scorer(fpr_score, neg_label=neg_label, pos_label=pos_label),
+    "tpr": make_scorer(tpr_score, pos_label=pos_label),
+}
+
+# %%
+# In addition, the original research [1]_ defines a custom business metric. We
+# call a "business metric" any metric function that aims at quantifying how the
+# predictions (correct or wrong) might impact the business value of deploying a
+# given machine learning model in a specific application context. For our
+# credit prediction task, the authors provide a custom cost-matrix which
+# encodes that classifying a a "bad" credit as "good" is 5 times more costly on
+# average than the opposite: it is less costly for the financing institution to
+# not grant a credit to a potential customer that will not default (and
+# therefore miss a good customer that would have otherwise both reimbursed the
+# credit and payed interests) than to grant a credit to a customer that will
+# default.
+#
+# We define a python function that weight the confusion matrix and return the
+# overall cost.
+import numpy as np
+
+
+def gain_cost_score(y, y_pred, neg_label, pos_label):
+    cm = confusion_matrix(y, y_pred, labels=[neg_label, pos_label])
+    # The rows of the confusion matrix hold the counts of observed classes
+    # while the columns hold counts of predicted classes. Recall that here
+    # we consider "bad" as the positive class (second row and column).
+    # Scikit-learn model selection tools expect that we follow a convention
+    # that "higher" means "better", hence the following gain matrix assigns
+    # negative gains (costs) to the two kinds of prediction errors:
+    # - a gain of -1 for each false positive ("good" credit labeled as "bad"),
+    # - a gain of -5 for each false negative ("bad" credit labeled as "good"),
```

Comment:
We should also mention that, given that our model is calibrated and our data set representative and large enough, we do not need to tune the threshold, but can safely set it to the cost ration 1/5, see Elkan Eq. (2).

https://arxiv.org/abs/1704.08979 mentions how to further reduce the parameters of the cost matrix.